### ISSUE TITLE
1.101.4 web w/o acceptsClassName: hoverStyle not working (regression)

### ISSUE THREAD

**Issue by mfazekas** [tamagui/tamagui#2773]
### Current Behavior

```tsx
const Button = styled("div", {
   ...
   hoverStyle: { backgroundColor: "lightgray", },
);
```

hoverStyle doesn't apply on 1.101.4 or later, tried up to 1.101.7.

https://github.com/tamagui/tamagui/assets/52435/c6ed48f4-b0b2-478d-8b23-6a93dedbce5d

### Expected Behavior

hoverStyle to apply as before. 

https://github.com/tamagui/tamagui/assets/52435/bf897484-6fca-4083-bbb6-b363fb1fcfba

### Tamagui Version

```markdown
1.101.4 and later (tested up to 1.101.7)
```

### Platform (Web, iOS, Android)

```markdown
Web
```

### Reproduction

```markdown
https://codesandbox.io/p/sandbox/amazing-leaf-zfszt4?file=%2Fsrc%2FApp.tsx%3A18%2C4
```

If I use `debug="verbose"` on 1.101.4 I do see:
```js
events {
    "events": null,
    "attachHover": false,
    "attachPress": false
}
```

on 1.101.3 I do see:
```js
events {
    "events": {},
    "attachHover": true,
    "attachPress": false
}
```

I can work around the issue by adding `acceptsClassName: true` 

### System Info

```markdown
n/a
```

**Comment by natew** [comment]
`styled('div')` is not supported by Tamagui, can you update the repro and make sure you're using the vanilla packages?

**Comment by mfazekas** [comment]
@natew updated the repo I could reproduce with a simple custom component. 

```jsx
function MyComponent({ style, onMouseEnter, onMouseLeave }) {
  console.log(" => style", style, onMouseEnter, onMouseLeave);
  return (
    <div
      style={{ width: 20, height: 20, ...style }}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    ></div>
  );
}

const MyStyledComponent = styled(MyComponent, {
  name: "MyStyledComponent",

  backgroundColor: "red",
  hoverStyle: {
    backgroundColor: "blue",
  },
});
```

This worked `1.101.3`, onMouseEnter and onMouseLeave was passed to the component and style was changed on hover.

On `1.104.2` onMouseEnter, onMouseLeave are not passed to the component.

**Comment by mfazekas** [comment]
Looking at the code:

https://github.com/tamagui/tamagui/blob/148e67ac5976110e9a405318a1db5265878ec69c/packages/web/src/createComponent.tsx#L287-L296

`!staticConfig.acceptsClassName` went away in new versions:

https://github.com/tamagui/tamagui/blob/13cfbf3388b2a9aab4999dc34472322996820ee0/code/core/web/src/createComponent.tsx#L287-L324

This seems to have changed in https://github.com/tamagui/tamagui/commit/fc4be94688f45ba497a25ba129348ef83cd9b408

**Comment by natew** [comment]
fix 6e23ee018616dc9bdae3a773c7fe1d74033ec0b8 for acceptsClassName on the third argument not being used.
